core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/iteration/IterateListSimultaneouslyUnitTest.java

```diff
@@ -0,0 +1,77 @@
+package com.baeldung.list.iteration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.data.util.StreamUtils;
+
+public class IterateListSimultaneouslyUnitTest {
+
+    final List<String> countryName = List.of("USA", "UK", "Germany", "India");
+    final List<String> countryCode = List.of("+1", "+44", "+49", "+91");
+
+    @Test
+    public void givenTwoLists_whenProcessedByZipping_thenGetJoinedDataFromBothCollections() {
+        List<String> processedList = StreamUtils.zip(countryName.stream(), countryCode.stream(),
+            (name, code) -> String.format("%s: %s", name, code))
+            .collect(Collectors.toList());
+        assertThat(processedList).containsExactly("USA: +1", "UK: +44", "Germany: +49", "India: +91");
+    }
+
+    @Test
+    public void givenTwoLists_whenIterateUsingIterator_thenGetJoinedDataFromBothCollections() {
+        Iterator nameIterator = countryName.iterator();
+        Iterator codeIterator = countryCode.iterator();
```

Comment:
Need to add the correct strong type here - `Iterator<String>`

Reply:
Done.